base/log_keys.go

```diff
@@ -0,0 +1,105 @@
+package base
+
+import (
+	"sync/atomic"
+)
+
+// LogKey is a bitfield of log keys.
+type LogKey struct {
+	flag uint32
+}
```

Comment:
Is there a reason this can't just be a type alias of LogKey to uint32, instead of a struct?

Reply:
I was trying to avoid all the type conversions (see example below), but the advantage of this is that all external constants/functions take in a typed `LogKey`, rather than mixing `uint32` and a `LogKey` struct. I'll change as it's beneficial for consistency's sake.

```gox
// Enable will enable the given logKey in keyMask.
func (keyMask *LogKey) Enable(logKey LogKey) {
	val := atomic.LoadUint32((*uint32)(keyMask))
	atomic.StoreUint32((*uint32)(keyMask), val|uint32(logKey))
}
```